test/server_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <sstream>
+#include <string>
+#include <iostream>
+
+#include "gtest/gtest.h"
+#include "server.hpp"
+#include <signal.h>
+
+
+TEST(ServerTest, PortTooBig) {
+	std::string addr = "0.0.0.0";
+	std::string port = "12312312";
+
+	http::server::server server_(addr, port);
+	EXPECT_FALSE(server_.isValid(addr, port));
+}
+
+TEST(ServerTest, PortIsNegative) {
```

Comment:
Does our program actually throw an exception when the port being passed is negative? I thought it was just handled the same way as ports that are too big -- the error is caught when the port number gets parsed and isValid will return false